### PR TITLE
Refactor the Housing Benefit parameter, variable and test files.

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Adjusted the format of the files.

Additionally:

Most parameters need 2023 and 2024 updates

Non-dependent deduction: age limit was 21 in model but it should be 18 - changed

Non-dep ded: income was accounted for as a weekly input which increased the deduction amount

Added `benunit_count_children.py`

### CODE REVIEW

**Comment by PavelMakarchuk** [comment]
Some more pressing outstanding questions that will be addressed in a separate PR:
2023 / 2024 amount for over state pension age filers - could not find the amounts in the references and will probably restructure

File: policyengine_uk/parameters/gov/dwp/housing_benefit/allowances/age_threshold/younger.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+description: A lower Houseing Benefit allowance amount is provided if both members of a couple are below this age threshold.
```
Comment (nikhilwoodruff) [inline policyengine_uk/parameters/gov/dwp/housing_benefit/allowances/age_threshold/younger.yaml]:
```suggestion
description: A lower Housing Benefit allowance amount is provided if both members of a couple are below this age threshold.
```

File: policyengine_uk/parameters/gov/dwp/housing_benefit/allowances/single/aged.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+description: The following personal allowance amount is provided to single filers over the pension age under the Housing Benefit.
+values:
+  2015-06-01: 181.00
+  2021-04-01: 191.15
+  2022-04-06: 197.10
```
Comment (nikhilwoodruff) [inline policyengine_uk/parameters/gov/dwp/housing_benefit/allowances/single/aged.yaml]: [Could we add the latest values (up to April 2024) from this Commons Library report?](https://commonslibrary.parliament.uk/research-briefings/cbp-9872/#:~:text=Inflation%2Dlinked%20benefits%20and%20tax,single%20people%20aged%20under%2025)

File: policyengine_uk/variables/gov/dwp/housing_benefit/applicable_income/housing_benefit_applicable_income.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        bi = parameters(period).gov.contrib.ubi_center.basic_income
+        TAX_COMPONENTS = ["income_tax", "national_insurance"]
+        benefits = add(benunit, period, BENUNIT_MEANS_TESTED_BENEFITS)
+        income = add(benunit, period, INCOME_COMPONENTS)
+        tax = add(benunit, period, TAX_COMPONENTS)
+        income += add(benunit, period, ["personal_benefits"])
+        if not bi.interactions.include_in_means_tests:
+            # Basic income is already in personal benefits, deduct if needed
+            income -= add(benunit, period, ["basic_income"])
+        income += add(benunit, period, ["tax_credits"])
```
Comment (nikhilwoodruff) [inline policyengine_uk/variables/gov/dwp/housing_benefit/applicable_income/housing_benefit_applicable_income.py]: This formula would be great to clean up a little bit- adding a few comments, and favouring assignment (post_tax_income = income - tax over income -= tax)?

File: policyengine_uk/variables/gov/dwp/housing_benefit/would_claim_HB.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+from policyengine_uk.model_api import *
+
+
+class would_claim_HB(Variable):
```
Comment (nikhilwoodruff) [inline policyengine_uk/variables/gov/dwp/housing_benefit/would_claim_HB.py]: We can decapitalise all variables I think

File: policyengine_uk/variables/household/demographic/benunit/benunit_count_children.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+from policyengine_uk.model_api import *
+
+
+class benunit_count_children(Variable):
+    value_type = int
+    entity = BenUnit
+    label = "Number of children in the Benefit Unit"
```
Comment (nikhilwoodruff) [inline policyengine_uk/variables/household/demographic/benunit/benunit_count_children.py]: Could we decapitalise all labels too? (except for those starting with a proper noun like Housing Benefit)

Comment (PavelMakarchuk) [reply policyengine_uk/variables/household/demographic/benunit/benunit_count_children.py]: Decapitalise the first word as well?
E.g.
```
"Number of children in the benefit unit"
```
or
```
"number of children in the benefit unit"
```

File: policyengine_uk/parameters/gov/dwp/housing_benefit/allowances/single/aged.yaml (hunk 2 above)
Comment (PavelMakarchuk) [reply policyengine_uk/parameters/gov/dwp/housing_benefit/allowances/single/aged.yaml]: Added but I would want to consolidate with the legislative source down the road

File: policyengine_uk/variables/household/demographic/benunit/benunit_count_children.py (hunk 5 above)
Comment (anth-volk) [reply policyengine_uk/variables/household/demographic/benunit/benunit_count_children.py]: Can't speak for Nikhil, but I believe so, as I believe he prefers lowercase in the app variable tree

Comment (nikhilwoodruff) [reply policyengine_uk/variables/household/demographic/benunit/benunit_count_children.py]: As a general rule of thumb I think you should be able to drop `label` as-is mid-sentence in 'X entity's VARIABLE_LABEL is Y'

**Review by nikhilwoodruff — APPROVED** [review]
Looks good to merge as-is (after label changes, though feel free to not have those in this PR as they're not that important) if we're doing the parameter updates outside this!

File: policyengine_uk/variables/household/demographic/benunit/benunit_count_children.py (hunk 5 above)
Comment (PavelMakarchuk) [reply policyengine_uk/variables/household/demographic/benunit/benunit_count_children.py]: That makes sense, I was going off of the idea that when we search for variables on the household side the labels just appear as VARIABLE_LABEL but I am fairly indifferent so I will change to lowercase

**Comment by PavelMakarchuk** [comment]
> Looks good to merge as-is (after label changes, though feel free to not have those in this PR as they're not that important) if we're doing the parameter updates outside this!

Great! All labels are changed and the existing parameters are up to date as of 04-2024 - will make structural changes outside of here

**Pull request merged**